.github/workflows/self-push.yml

```diff
@@ -10,73 +10,43 @@ on:
       - "tests/**"
       - ".github/**"
       - "templates/**"
-  # pull_request:
   repository_dispatch:
 
-
 jobs:
   run_tests_torch_gpu:
-    runs-on: [self-hosted, gpu, single-gpu]
+    runs-on: [self-hosted, docker-gpu, single-gpu]
+    container:
+      image: pytorch/pytorch:1.8.0-cuda11.1-cudnn8-runtime
+      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
```

Comment:
Do you need --gpus all if you want to run only on a single GPU?

You can do --gpus 0 and then remove relying on CUDA_VISIBLE_DEVICES: 0